docs/IR.md

```diff
@@ -1,127 +1,382 @@
 =========
 
-ONNX is an open specification that consists of the following components:
+__Purpose__
+
+This document contains the normative specification of the semantics of ONNX. The .proto and .proto3 files found under the ‘onnx’ folder form the normative specification of its syntax. Commentary found in the .proto and .proto3 files are intended to improve readability of those files, but are not normative if they conflict with this document. Such conflicts should be reported as documentation bugs.
 
-1)  A definition of an extensible computation graph model
+__Notes on model validation__
 
-2)  Definitions of built-in operators and standard data types
+A [Python-based tool](../onnx/checker.py) is available to perform general validation of models against this specification.
 
 __Notes on language in this and all related documents__:
 
 1. The use of SHOULD, MUST, MAY and so on in this document is consistent with [RFC 2119](https://www.ietf.org/rfc/rfc2119.txt).
 
 2. The use of 'list' shall denote an ordered collection of items, 'set' shall denote an unordered collection of unique elements, and 'bag' an unordered collection of possibly non-unique elements.
 
-Extensible computation graph model
-----------------------------------
+## Components
+
+ONNX is an open specification that consists of the following components:
+
+1)  A definition of an extensible computation graph model.
+
+2)  Definitions of standard data types.
+
+3)  Definitions of built-in operators.
+
+Of these, #1 and #2 are covered herein; the built-in operators are covered separately in documents listed at the end of this.
+
+There are two official ONNX variants; the main distinction between the two is found in the supported types and the default operator sets. The neural-network-only __ONNX__ variant recognizes only tensors as input and output types, while the Classical Machine Learning extension, __ONNX-ML__, also recognizes sequences and maps. __ONNX-ML__ extends the __ONNX__ operator set with  ML algorithms that are not based on neural networks.
+
+## Runtime Agnostic
+
+ONNX does not pre-suppose or imply any particular method of runtime implementation. 
+
+For example, an implementation may consist of a rich runtime which interprets the model; it may be a code generator that translates the model in its entirety to executable code for some target programming language; it may be a hardware implementation; it may be a combination of two or three of those. 
+
+Nothing in this specification should be construed as advocating one implementation approach over any other; any comments on the inner workings of concrete implementations are to be interpreted as examples.
+
+## ONNX Versioning
+
+Versioning features in several places in ONNX -- the IR specification itself, the version of a model, and the version of an operator set. Futhermore, each individual operator indicates which version of its containing operator set it was introduced or stabilized in.
```

Comment:
Shall we also talk about release version here?

Reply:
I guess we can explain more in Versioning.md

Reply:
That makes sense. I intend to tackle Versioning.md next, by the way.